components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt

```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
 @Suppress("Deprecation")
 @Parcelize
 data class DownloadState(
-    val url: String,
-    val fileName: String? = null,
-    val contentType: String? = null,
-    val contentLength: Long? = null,
-    val userAgent: String? = null,
-    val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
-    val referrerUrl: String? = null,
-    val skipConfirmation: Boolean = false,
-    val id: Long = Random.nextLong(),
-    val sessionId: String? = null
+        val url: String,
+        val fileName: String? = null,
+        val contentType: String? = null,
+        val contentLength: Long? = null,
+        var currentBytesCopied: Long = 0,
+        @GuardedBy("context") var status: DownloadJobStatus = DownloadJobStatus.ACTIVE,
```

Comment:
Let's add docs :)
I think we can remove the `@GuardedBy` annotation

Reply:
We want this values to be `val` as we want them to be immutable.

Reply:
Do you think it's ok to set it as Active right away? It wouldn't let me not set it to anything like before.